config.go

```diff
@@ -10,8 +10,12 @@ import (
 	"github.com/aws/aws-sdk-go/aws/session"
 )
 
+type SessionProviderFunc func(c Config) (*session.Session, error)
```

Comment:
Can you add a gofmt standard comment for this function in the form of

// SessionProviderFunc ... 
so that it will show up in the GoDoc

Additionally, will you include a simple example or usage instructions on how someone might leverage this function to create a custom session set up. (You can even just use the code in the test file)

After that the PR is good to go

Reply:
@jdoklovic are you still interested in finalizing this PR? If you are too busy, I can take it over. I greatly appreciate your contribution so far

Reply:
@qhenkart Yeah, I'll add the requested changes today and update the PR